git_machete/github.py

```diff
@@ -123,6 +123,8 @@ def fire_github_api_get_request(method: str, url: str, token: Optional[str], req
 
         if 200 <= response.status < 300:
             return parsed_response_body
+        elif response.status == 422 and parsed_response_body.get("message") == "Validation Failed":
+            raise MacheteException("Cannot create pull request on untracked branch. \nPlease push your changes before making pull request.")
```

Comment:
Hmmm putting this specific message to the generic `fire_github_api_get_request` method is questionable... note that this method is used not only for creating PRs, but also for a few other types of requests.

In general, I'd rather throw a dedicated exception from `fire_github_api_get_request`, to be caught & interpreted by each high-level function.